pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -154,6 +167,17 @@ def on_train_batch_end(
 
         trainer.logger.log_metrics(logs, step=trainer.global_step)
 
+    @staticmethod
+    def _get_gpu_ids(device_ids: List[int]) -> str:
+        """Get real GPU IDs"""
+        cuda_visible_devices = os.getenv('CUDA_VISIBLE_DEVICES', default=None)
+        if cuda_visible_devices is None:  # `CUDA_VISIBLE_DEVICES` is unset
+            gpu_ids = ','.join(map(str, device_ids))
+        else:
+            cuda_visible_devices = cuda_visible_devices.split(',')
+            gpu_ids = ','.join(map(cuda_visible_devices.__getitem__, device_ids))
+        return gpu_ids
```

Comment:
I prefer to use the if-else switchs (`default=None`) for:

1. It could make an unnecessary module import and an unnecessary function call if set `default=','.join(map(str, range(torch.cuda.device_count())))` when `CUDA_VISIBLE_DEVICES` is set.
2. For testing, `torch.cuda.device_count()` depends on the runtime environment of the CI test container. It could be hard to write unit tests.

Reply:
1. python only imports once and then caches it. repeated calls to torch.cuda... have no impact
2. `torch.cuda.device_count` is always available, even on a machine without GPUs (returns 0). We can write the test for a GPU machine and just only run it if gpus are available :)